operations/mimir-mixin/alerts/autoscaling.libsonnet

```diff
@@ -1,27 +1,34 @@
 (import 'alerts-utils.libsonnet') {
-  groups+: if !$._config.autoscaling.querier_enabled then [] else [
-    local regexMatching = function(strings=[]) std.join('|', std.map(function(s) '(%s)' % s, strings));
+  local anyEnabled(components) = (
+    // NOTE(jhesketh): std.any is only available in jsonnet > 0.19. Instead, use .count for now
+    if std.count([(component['enabled']) for component in components], true) > 0 then (
+      true
+    ) else (
+      false
+    )
+  ),
+  groups+: if !anyEnabled($._config.autoscaling) then [] else [
```

Comment:
I would keep it simple and always configure the alert, regardless whether autoscaling is enabled or not. If autoscaling is not enabled, then the metric `kube_horizontalpodautoscaler_status_condition` won't exist and alerting rule will not evaluate. That's an approach we take for other alerts too, which are based on features you may or may not have enabled.